components/rollup_node/README.md

```diff
@@ -1,62 +1,109 @@
 # Rollup Node
 
-The consensus module of Optimistic Ethereum.
+The consensus-layer of Optimism.
+
+This design mirrors the separation of L1 consensus- and execution-layer to L2,
+for maximum compatibility and minimal complexity.
 
 ## Summary
 
-The Rollup Node is a consensus client that determines the latest state of the rollup. It reads state from L1, and possibly the sequencer, to compute the L2 state. The [block generation spec][block-gen] defines the rules by which L1 state is converted into L2 blocks.
+The Rollup Node is a consensus client that tracks the rollup:
+- Fork-choice: following L1
+- Input Data: deposits and sequencer data extracted from L1
+- Latest blocks: optimistically via L2 p2p, overrideable by L1
+
+The input data and forkchoice information collected from L1 (and optimistically via p2p)
+are then forwarded to the execution-layer, also known as "execution engine",
+via the standard [Engine API][engine-api].
+
+The design is minimalist:
+- The rollup consensus-layer is largely derived from L1 properties, enabling a simple and stateless L2 consensus.
+- The rollup execution-layer is largely shared with L1, equivalent in all ways except the addition of rollup-deposits.
+
 
 ## Types of Rollup Nodes
+
 There are two primary classes of rollup nodes, corresponding to different configurations of the software stack.
 
-1. **Sequencer**: Accepts user-sent and deposit transactions, orders them into L2 blocks, and submits them to L1 in batches. Provides RPC access to pending L2 blocks in advance of L1 submission.
-2. **Verifier**: Watches L1 feeds, reconstructs L2 block inputs, and inserts them into the EE to determine the canonical L2 state. May also track pending sequencer state, in some configurations.
+1. **Sequencer**: Accepts user-sent and deposit transactions, orders them into L2 blocks, and submits them to L1 in batches.
+2. **Verifier**: Watches L1 feeds, reconstructs L2 block inputs, and inserts them into the EE to determine the canonical L2 state.
+
+**Both** update their coupled execution-engine with L2 blocks and fork-choice instructions, to maintain their view of the rollup state.
+
+Optionally the sequencer(s) may share the L2 blocks via the L2 p2p network before the L1 can confirm them,
+which verifiers can optionally use to sync to the very tip (marked "unsafe" in JSON-RPC) of the rollup.
+This enables permissionless L1-like latency for users, but not canonical until available on the L1. 
```

Comment:
"finalized" would be confusing with both the other field of the forkchoice instruction ("finalizedBlockHash"), as well as the event of finality, which is not what matters here. The "latest" should be safe enough to use. It might still reorg, but is unlikely to. I'll change the wording to clarify this.